### PR TITLE
Branch6.2

### PR DESCRIPTION
**Pull request by vbortnyk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int num = 3;
+        num = length * 31 * num;
+        num = width * 31 * num;
+        num = color.hashCode() * 31 * num;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: what happens, if color == null ?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (this.getClass().equals(obj.getClass())) {
+            Rectangle rect = (Rectangle) obj;
+            if (width.equals(rect.width)
+                    && length.equals(((Rectangle) obj).length)
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: do we need to cast obj again if we have already rect object?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+                    && color.equals(rect.color)
+                    && this.hashCode() == rect.hashCode()) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: do we need this check?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (vbortnyk) [reply src/main/java/core/basesyntax/Rectangle.java]: edited

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (vbortnyk) [reply src/main/java/core/basesyntax/Rectangle.java]: edited

File: src/main/java/core/basesyntax/Rectangle.java (hunk 3 above)
Comment (vbortnyk) [reply src/main/java/core/basesyntax/Rectangle.java]: edited

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**